### PR TITLE
[NEW] Added Room Settings Functions

### PR DESCRIPTION
**Pull request by samrmur**
**Information**
This PR adds the following features:

- Leave Room
- Get Room Info
- Get Roles from Subscription
- Set Room Topic
- Set Room Announcement
- Set Room Description
- Set Room to be Read-Only
- Set Room Type
- Set Room Join Code/Password
- Open/Close Subscription
- Archive/Unarchive Room

I have added test cases for all the above features

### CODE REVIEW

**Comment by rafaelks** [comment]
@samrmur Nice one!!! We're going to review this one soon. 👍

**Comment by rafaelks** [comment]
@samrmur I'm sorry the delay... can you fix the conflicts please?

**Comment by samrmur** [comment]
@rafaelks Sure, sorry I saw this late, will fix conflicts as soon as possible

File: core/src/main/kotlin/chat/rocket/core/internal/rest/ChatRoom.kt [hunk 1]
```diff
@@ -70,10 +82,143 @@ suspend fun RocketChatClient.joinChat(roomId: String): Boolean = withContext(Com
...
+    val payload = ChatRoomDescriptionPayload(roomId, description)
+    val adapter = moshi.adapter(ChatRoomDescriptionPayload::class.java)
+    val payloadBody = adapter.toJson(payload)
+
+    val body = RequestBody.create(MEDIA_TYPE_JSON, payloadBody)
+
+    val url = requestUrl(restUrl, getRestApiMethodNameByRoomType(roomType, "setDescription")).build()
+    val request = requestBuilder(url).post(body).build()
+
+    System.out.println(request.body().toString())
```
Comment (leonardoaramaki) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/ChatRoom.kt]: Can you remove this log?

File: core/src/main/kotlin/chat/rocket/core/internal/rest/ChatRoom.kt [hunk 2]
```diff
@@ -5,9 +5,9 @@ import chat.rocket.common.model.RoomType
 import chat.rocket.common.model.User
 import chat.rocket.core.RocketChatClient
 import chat.rocket.core.internal.RestResult
-import chat.rocket.core.internal.model.ChatRoomJoinPayload
-import chat.rocket.core.internal.model.ChatRoomPayload
+import chat.rocket.core.internal.model.*
```
Comment (filipedelimabrito) [inline core/src/main/kotlin/chat/rocket/core/internal/rest/ChatRoom.kt]: This wildcard will fail the code formatting.

File: core/src/test/kotlin/chat/rocket/core/internal/rest/ChatRoomTest.kt [hunk 3]
```diff
@@ -115,4 +115,241 @@ class ChatRoomTest {
             assertTrue(result)
         }
     }
+<<<<<<< HEAD
```
Comment (filipedelimabrito) [inline core/src/test/kotlin/chat/rocket/core/internal/rest/ChatRoomTest.kt]: We just need to remove it. But I won't bother you and will remove it myself.
Really thank you @samrmur!
Thanks for this pull request and your patience! ❤️

**Pull request merged**